src/Host.cpp

```diff
@@ -3533,18 +3533,36 @@ bool Host::resetBackgroundImage(const QString &name)
 // use it WITHOUT loading a file - at least for the TConsole::importMap(...)
 // case that may need to create a map widget before it loads/imports a
 // non-default (last saved map in profile's map directory.
-void Host::createMapper(bool loadDefaultMap)
+void Host::toggleVisibilityOfOrCreateMapper(const bool loadDefaultMap)
 {
     auto pMap = mpMap.data();
     if (pMap->mpMapper) {
-        bool visStatus = mpMap->mpMapper->isVisible();
-        if (pMap->mpMapper->parentWidget()->inherits("QDockWidget")) {
-            pMap->mpMapper->parentWidget()->setVisible(!visStatus);
-        }
-        pMap->mpMapper->setVisible(!visStatus);
+        toggleMapperVisibility();
         return;
     }
 
+    createMapper(loadDefaultMap);
+}
+
+void Host::toggleMapperVisibility()
+{
+    auto pMap = mpMap.data();
+    bool visStatus = mpMap->mpMapper->isVisible();
+    if (pMap->mpMapper->parentWidget()->inherits("QDockWidget")) {
```

Comment:
This should be a method of its own, `if (pMap->mpMapper->floatingWidget) {`...

Reply:
Tracing it back through the history I see it came from Chris in 2013 in 5549b3690b0cf5f3fd8099708318ccc9a768f049 but that looks as though it was something tagged onto an unrelated commit - probably to properly hide the floating/dockable widget introduced in 41e0cbbed58f518321328e236429ce4d7097c6d6 (Mudlet 1.2.9-pre4).

Reply:
Done.